src/lib/connectors/shared/remote-debugging-connector.ts

```diff
@@ -326,7 +326,12 @@ export class Connector implements IConnector {
             };
         }
 
-        if (hasAttributeWithValue(data.element, 'link', 'rel', 'icon')) {
+        const hasFaviconLinkTag = hasAttributeWithValue(data.element, 'link', 'rel', 'icon');
+        // If favicon is not present in the html, CDP will send a request to get favicon from the root directory.
+        // We should set the tag to true in this case to avoid fetching favicon twice.
+        const hasFaviconInRootDir = data.request.url.endsWith('/favicon.ico');
```

Comment:
I don't think this is right. I could very well have the following url `https://sonarwhal.com/images/favicon.ico` and the favicon will not be in the root dir.
We should add a test for this case too.